### ISSUE TITLE
no-unused-keys: allow ignore-per-file

### ISSUE THREAD

**Issue by alxndrsn** [intlify/eslint-plugin-vue-i18n#668]
**What rule do you want to change?**

`no-unused-keys`

**Does this change cause the rule to produce more or fewer warnings?**

More warnings.

**How will the change be implemented? (New option, new default behavior, etc.)?**

Option `ignores`: option to supply object instead of array, restricting lists of rules to groups of files.

**Please provide some example code that this change will affect:**

```js
ignores: {
  '**/SomeComponent.vue': [
    '/^dynamic\\.key\\.particular_to_this_file\\./',
  ],
  '**/*': [
    'very.special.key',
  ],
},
```

**What does the rule currently do for this code?**

Currently `ignores` rules have to be global, e.g.

```js
ignores: [
  '/^dynamic\\.key\\.particular_to_this_file\\./',
  'very.special.key',
],
```

**What will the rule do after it's changed?**

Allow more fine-grained ignore rules.
